samcli/lib/providers/provider.py

```diff
@@ -57,10 +59,15 @@ def __init__(self):
         self.result = {"Function": [], "Layer": []}
 
     def add_function(self, function):
+        if function.inlinecode and function.inlinecode is not None:
+            return
         self.result.get("Function").append(function)
 
     def add_functions(self, functions):
-        self.result.get("Function").extend(functions)
+        for function in functions:
+            if function.inlinecode and function.inlinecode is not None:
```

Comment:
You could maybe call the `add_function` above here

Reply:
makes sense!